src/utils/select-rows.js

```diff
@@ -129,51 +154,88 @@ export function selectRows(rowInfo, options) {
     return null;
 }
 
+/**
+ * Aggregate row information using `rowAggregate` options.
+ * Data formats are changed from [number, Object] to [number[], Object[]] for aggregated rows.
+ * @param {object[]} rowInfo The original/full default-ordered rowInfo array.
+ * @param {object} rowAggregate The track option object for aggregating rows.
+ * @returns {object[]} The aggregated `rowInfo`.
+ */
+export function getAggregatedRowInfo(rowInfo, rowAggregate) {
+    let aggregatedRowInfo = Array.from(rowInfo.entries());
+    if(rowAggregate && rowAggregate.length > 0) {
+        const aggregateOptions = rowAggregate;
+        aggregateOptions.forEach(d => {
+            const { field, type, oneOf } = d;
+            if(type === "nominal") {
+                oneOf.forEach(one => {
+                    const matchings = aggregatedRowInfo.filter(t => t[1][field] === one);
+                    const notMatchings = aggregatedRowInfo.filter(t => t[1][field] !== one);
+                    const matchingIndices = matchings.map(t => t[0]);
+                    const matchingRowInfos = matchings.map(t => t[1]);
+
+                    if(matchings?.length == 0) return;
+
+                    aggregatedRowInfo = notMatchings;
+                    aggregatedRowInfo.push([matchingIndices, matchingRowInfos]);
+                });
+            }
+        });
+    }
+    return aggregatedRowInfo;
+}
+
 /**
  * Generate an array of highlighted row indices based on conditions.
  * @param {object[]} rowInfo The original/full default-ordered rowInfo array.
  * @param {string} field The attribute on which to search.
  * @param {string} type The data type contained in the field value.
  * @param {string|array} conditions The conditions to check for highlighting.
+ * @param {object} options The track options object, containing aggregation options.
  * @returns {number[]} The array of highlit indices.
  */
-export function highlightRowsFromSearch(rowInfo, field, type, conditions) {
+export function highlightRowsFromSearch(rowInfo, field, type, conditions, options) {
+    // Aggregation rows in advance to apply highlighting to the aggregated rows.
+    const aggregatedRowInfo = getAggregatedRowInfo(rowInfo, options.rowAggregate);
+    const aggFunction = options.rowInfoAttributes?.find(d => d.field === field)?.aggFunction;
```

Comment:
Can we differentiate variables containing names of functions from variables that are functions themselves by appending `Name`, so here it would be `const aggFunctionName`? Or maybe `const aggFuncName` for short